hazelcast/src/main/java/com/hazelcast/map/impl/MapMigrationAwareService.java

```diff
@@ -184,7 +193,8 @@ private void removeOrRegenerateNearCacheUuid(PartitionMigrationEvent event) {
     @Override
     public void rollbackMigration(PartitionMigrationEvent event) {
         if (DESTINATION == event.getMigrationEndpoint()) {
-            removeRecordStoresHavingLesserBackupCountThan(event.getPartitionId(), event.getCurrentReplicaIndex());
+            removeRecordStoresHavingLesserBackupCountThan(event.getPartitionId(),
```

Comment:
Do we need to remove the reservations here if the member is no longer the replica after rollback, same as for record stores?

Reply:
you are right i will update PR